src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -1597,15 +1582,13 @@ parse_reply_time (const bson_t *reply, op_time_t *op_time)
 }
 
 
-#define ASSERT_OP_TIMES_EQUAL(_a, _b)                             \
-   if ((_a).t != (_b).t || (_a).i != (_b).i) {                    \
-      fprintf (stderr,                                            \
-               #_a " (%d, %d) does not match " #_b " (%d, %d)\n", \
-               (_a).t,                                            \
-               (_a).i,                                            \
-               (_b).t,                                            \
-               (_b).i);                                           \
-      abort ();                                                   \
+#define ASSERT_OP_TIMES_EQUAL(_a, _b)                              \
+   if ((_a).t != (_b).t || (_a).i != (_b).i) {                     \
+      test_error (#_a " (%d, %d) does not match " #_b " (%d, %d)", \
+                  (_a).t,                                          \
+                  (_a).i,                                          \
+                  (_b).t,                                          \
+                  (_b).i);                                         \
```

Comment:
I would suggest that we wrap multiline macros such as this in a `do { ... } while (0)`, since as written it can behave unexpectedly within an `if/else`, for example, if we have the following code, we will get a syntax error:
```
 11 #define FOO(X) \
 12     if (X) { \
 13         foo(X); \
 14         bar(X); \
 15     }
...
 25 int main(int argc, char **argv) {
 26     if (argc > 2)
 27         FOO(argc);
 28     else
 29         puts("hello");
 30 }
```
Syntax error:
```
2022-12-16 9:04:05 [~/src/test]$ gcc dowhile.c
dowhile.c:28:5: error: expected expression
    else
    ^
1 error generated.
```

However, if we wrap in a `do/while(0)`, the macro works as expected:
```
 17 #define BAR(X) \
 18     do { \
 19         if (X) { \
 20             foo(X); \
 21             bar(X); \
 22         } \
 23     } while (0)
 24
 25 int main(int argc, char **argv) {
 26     if (argc > 2)
 27         BAR(argc);
 28     else
 29         puts("hello");
 30 }
```
Compiles as expected:
```
2022-12-16 9:04:37 [~/src/test]$ gcc dowhile.c
2022-12-16 9:04:38 [~/src/test]$
```

Reply:
Good spot. Updated accordingly using the pattern [described in your other comment](https://github.com/mongodb/mongo-c-driver/pull/1164#discussion_r1050829750).